src/math.hpp

```diff
@@ -80,6 +80,28 @@ int signum(T x) {
     return (x>T(0)) - (x<T(0));
 }
 
+// Return minimum of the two values
+// Note: returns first argument in case of equality
+template <typename T>
+T min(const T& lhs, const T& rhs) {
+    return rhs<lhs? rhs: lhs;
+}
+
+// Return maximum of the two values
+// Note: returns second argument in case of equality
+template <typename T>
+T max(const T& lhs, const T& rhs) {
+    return lhs<rhs? lhs: rhs;
```

Comment:
This also returns minimum.

Reply:
Is there a good reason not to wrap `std::min` or `std::fmin` (for stronger guarantees) instead?

Reply:
I am not too concerned about stronger guarentees. So we could use `std::min`... I have just used a ternary (with a test to make sure I got it right).